src/rpc/rawtransaction.cpp

```diff
@@ -122,6 +122,63 @@ void TxToJSON(const CTransaction& tx, const uint256 hashBlock, UniValue& entry)
     }
 }
 
+void CoinsByScriptToJSON(const CCoinsByScript& coinsByScript, int nMinDepth, UniValue& vObjects, std::vector<std::pair<int, unsigned int>>& vSort, bool fIncludeHex)
+{
+    BOOST_FOREACH(const COutPoint &outpoint, coinsByScript.setCoins)
+    {
+        CCoins coins;
+        if (nMinDepth == 0)
+        {
+            LOCK(mempool.cs);
+            CCoinsViewMemPool view(pcoinsTip, mempool);
+            if (!view.GetCoins(outpoint.hash, coins))
+                continue;
+            mempool.pruneSpent(outpoint.hash, coins); // TODO: this should be done by the CCoinsViewMemPool
+        }
+        else if (!pcoinsTip->GetCoins(outpoint.hash, coins))
+            continue;
```

Comment:
Inconsistent behaviour for UTXOs created in mined transactions vs in the mempool. We should probably consider mempool spends the same for both mempool-created UTXOs and confirmed-tx-created UTXOs.